compiler/rustc_hir_analysis/src/astconv/errors.rs

```diff
@@ -102,6 +101,7 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
         &self,
         all_candidates: impl Fn() -> I,
         ty_param_name: &str,
+        ty_param_def_id: Option<LocalDefId>,
```

Comment:
nit: can you maybe combine `ty_param_name` and `ty_param_def_id` into a new type instead of passing them separately?

Reply:
done